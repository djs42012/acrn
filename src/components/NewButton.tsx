import { PressableProps, Pressable, Text, StyleSheet } from "react-native";
import React from "react";

interface Props extends PressableProps {
  text?: string
}

const defaultButtonStyle = StyleSheet.create({
  pressable: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 20,
    elevation: 4,
    paddingVertical: 12,
    paddingHorizontal: 32,
    backgroundColor: 'blue',
    marginVertical: 5
  },
  text: {
    fontWeight: 'bold',
    color: 'white',
    fontSize: 14
  }
})

function NewButton(props: Props) {
  const { text } = props;
  return (
    <Pressable accessibilityRole="button" style = {props.style? props.style.pressable: defaultButtonStyle.pressable} {...props}>
      <Text style = {props.style? props.style.text: defaultButtonStyle.text }>{text ? text: 'Submit'}</Text>
    </Pressable>
  );
}

export default NewButton;

