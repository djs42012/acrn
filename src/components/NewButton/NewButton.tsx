import { PressableProps, Pressable, Text } from "react-native";
import React from "react";

interface Props extends PressableProps {
  text?: string
}

function NewButton(props: Props) {
  const { text } = props;
  return (
    <Pressable accessibilityRole="button" {...props}>
      <Text>{text}</Text>
    </Pressable>
  );
}

export default NewButton;
