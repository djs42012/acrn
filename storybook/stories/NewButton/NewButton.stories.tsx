import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../CenterView';
import NewButton from '../../../src/components/NewButton/NewButton';

storiesOf('NewButton', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('New Button', () => (
    <NewButton
      onPressIn={() => console.log('onPressIn')}
      onPress={() => console.log('onPress')}
      onPressOut={() => console.log('onPressOut')}
      onLongPress={() => console.log('onLongPress')}
    />
  ));
