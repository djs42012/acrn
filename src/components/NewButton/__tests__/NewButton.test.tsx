import React from 'react';
import NewButton from '../NewButton';
import { render } from '@testing-library/react-native';

describe('Accessible Text Input', () => {
  // let button: RenderAPI | ReactTestInstance;
  const props = {
    text: 'New Button',
  };
  // beforeEach(() => {
  //   button = render(<NewButton {...props} />);
  // })
  test('renders input text', () => {
    const { getByText } = render(<NewButton {...props} />);
    expect(getByText('New Button')).toBeTruthy();
  });
  // xtest('button is pressable', () => {
  //   const onClick = jest.fn();
  //   fireEvent(button, onClick);
  //   fireEvent;
  // });
});
