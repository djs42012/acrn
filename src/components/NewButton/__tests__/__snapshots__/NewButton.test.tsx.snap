// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`Accessible NewButton accessibility role is button 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="New Button"
>
  <Text>
    New Button
  </Text>
</View>
`;

exports[`Accessible NewButton button is pressable 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="New Button"
>
  <Text>
    New Button
  </Text>
</View>
`;

exports[`Accessible NewButton renders input text 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="New Button"
>
  <Text>
    New Button
  </Text>
</View>
`;

exports[`Accessible Text Input accessibility role is button 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="New Button"
>
  <Text>
    New Button
  </Text>
</View>
`;

exports[`Accessible Text Input button is pressable 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="New Button"
>
  <Text>
    New Button
  </Text>
</View>
`;

exports[`Accessible Text Input renders input text 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="New Button"
>
  <Text>
    New Button
  </Text>
</View>
`;
